cvat/apps/events/views.py

```diff
@@ -0,0 +1,151 @@
+                description="Filter events by user ID"),
+            OpenApiParameter('from', location=OpenApiParameter.QUERY, type=OpenApiTypes.DATETIME, required=False,
+                description="Filter events after the datetime"),
+            OpenApiParameter('to', location=OpenApiParameter.QUERY, type=OpenApiTypes.DATETIME, required=False,
+                description="Filter events before the datetime"),
+            OpenApiParameter('filename', description='Desired output file name',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR, required=False),
+            OpenApiParameter('action', location=OpenApiParameter.QUERY,
+                description='Used to start downloading process after annotation file had been created',
+                type=OpenApiTypes.STR, required=False, enum=['download']),
+            OpenApiParameter('query-id', location=OpenApiParameter.QUERY, type=OpenApiTypes.STR, required=False,
+                description="ID of query request that need to check or download"),
+        ],
+        responses={
+            '200': OpenApiResponse(description='Download of file started'),
+            '201': OpenApiResponse(description='CSV log file is ready for downloading'),
+            '202': OpenApiResponse(description='Creating a CSV log file has been started'),
+        })
+    @action(detail=False, methods=['GET', 'POST'])
+    def events(request):
+        if request.method == 'POST':
+            serializer = ClientEventsSerializer(data=request.data)
+            serializer.is_valid(raise_exception=True)
+
+            org = request.iam_context['organization']
+            org_id = org.id if org else None
+
+            send_time = datetime_parser.isoparse(serializer.data["send_timestamp"])
+            receive_time = datetime.datetime.now(datetime.timezone.utc)
+            time_correction = receive_time - send_time
+            last_timestamp = None
+
+            for event in serializer.data["events"]:
+                event['user'] = request.user.id
+                timestamp = datetime_parser.isoparse(event['timestamp'])
+                if last_timestamp:
+                    t_diff = timestamp - last_timestamp
+                    if t_diff.seconds < EventsViewSet.TIME_THRESHOLD:
+                        payload = event.get('payload', {})
+                        if payload:
+                            payload = json.loads(payload)
+
+                        payload['working_time'] = t_diff.microseconds // 1000
+                        event['payload'] = json.dumps(payload)
+
+                last_timestamp = timestamp
+                event['timestamp'] = str((timestamp + time_correction).timestamp())
+                event['source'] = 'client'
+                event['organization'] = org_id
+                message = JSONRenderer().render(event).decode('UTF-8')
+                jid = event.get("job")
+                tid = event.get("task")
+                if jid:
```

Comment:
I don't think that we need clogger at all. What do you think?

Reply:
deleted